openapi.yml

```diff
@@ -0,0 +1,337 @@
+                                format: date
+      tags:
+        - menu
+  /enums/canteens.json:
+    get:
+      summary: To get all available canteens
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Canteen'
+      tags:
+        - static
+  /enums/languages.json:
+    get:
+      summary: To get all available languages
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Language'
+      tags:
+        - static
+  /enums/labels.json:
+    get:
+      summary: To get all available labels
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Label'
+      tags:
+        - static
+components:
+  schemas:
+    Canteen:
+      type: object
+      description: Information about a canteen
+      properties:
+        enum_name:
+          type: string
+          description: Internal identifier for a canteen
+          example: MENSA_GARCHING
+        name:
+          type: string
+          description: Readable name for a canteen
+          example: Mensa Garching
+        location:
+          $ref: '#/components/schemas/Location'
+        canteen_id:
+          type: string
+          description: Identifier for the canteen, as it has to be used for accessing data.
+          example: mensa-garching
+        queue_status:
+          type: string
+          description:
+            URL to another API, where data about the queue length of the canteen
+            can be accessed. Is null, when not available
+          example: https://mensa.liste.party/api
+    Location:
+      type: object
+      description: A location description including the readable address and coordinates
+      properties:
+        address:
+          type: string
+          description: Readable address consisting of street, house number and city
+          example: Boltzmannstraße 19, Garching
+        latitude:
+          type: number
+          example: 48.268132
+        longitude:
+          type: number
+          example: 11.672263
+    Language:
+      type: object
+      description: A language, that is supported by the API
+      properties:
+        name:
+          type: string
+          description: Identifier of the language
+          example: DE
+        base_url:
+          type: string
+          description:
+            Additional base path that is appended to the base url, in order to
+            access data for this language
+          example: en/
```

Comment:
```suggestion
          example: /
```
Wrong base path?

Reply:
I added the english one here as example, to have something to show, as the german `base_path` is currently given as `""` (empty string).
Actually, I think this could be changed, to have it in the same way (regarding leading/trailing slashes).
However, I'm not sure which version is the best.
Currently the format is basically: `{API_URL}{base path}{action}`
With:
API_URL: includes `/` at the end
base_url: includes `/` at the end (not the beginning)
action: should not include `/` at the beginning (due to base_path/API_URL already providing it)

What maybe a bit confusing here, is that sometimes the `/` in front of the action comes from the API_URL (in german case with base_path=`""`, and sometime from the base path (in english with base_path=`"en/"`. 
Do you think, we should change that, or just document it properly?

Reply:
In this case I think we should change the example here to:
```
    Language:
      type: object
      description: A language, that is supported by the API
      properties:
        name:
          type: string
          description: Identifier of the language
          example: EN
        base_url:
          type: string
          description:
            Additional base path that is appended to the base url, in order to
            access data for this language
          example: en/
          type: string
          description: Readable name of the language in the given language
          example: English
        flag:
          type: string
          description: Emoji flag symbol for the language
          example: 🏴󠁧󠁢󠁥󠁮󠁧󠁿
```
So we have a full fledged example.
And then we included something like: "Due to backwards compatibility reasons the `base_url` for German is an empty string."

Then we don't have to play arround with all those `\`.

Reply:
I agree, that sounds like a valid example, that I already included